src/main/java/cloud/fogbow/ras/core/plugins/interoperability/cloudstack/CloudStackCloudUtils.java

```diff
@@ -49,61 +49,77 @@ public static String doRequest(@NotNull CloudStackHttpClient cloudStackHttpClien
         }
     }
 
-    // TODO (chico) - move to the CloudstackCloudUtil
+    /**
+     * Wait and process the Cloudstack asynchronous response in its asynchronous life cycle.
+     * @throws FogbowException
+     */
     public static String waitForResult(@NotNull CloudStackHttpClient client,
                                        String cloudStackUrl,
                                        String jobId,
                                        @NotNull CloudStackUser cloudStackUser)
             throws FogbowException {
-        CloudStackQueryAsyncJobResponse queryAsyncJobResult = getAsyncJobResponse(client,
-                cloudStackUrl, jobId, cloudStackUser);
-
-        if (queryAsyncJobResult.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
-            for (int i = 0; i < MAX_TRIES; i++) {
-                queryAsyncJobResult = getAsyncJobResponse(client, cloudStackUrl, jobId, cloudStackUser);
-                if (queryAsyncJobResult.getJobStatus() != CloudStackQueryJobResult.PROCESSING) {
-                    return processJobResult(queryAsyncJobResult, jobId);
-                }
-                try {
-                    Thread.sleep(ONE_SECOND_IN_MILIS);
-                } catch (InterruptedException e) {
-                    throw new FogbowException();
-                }
+
+        int countTries = 0;
+        CloudStackQueryAsyncJobResponse response = getAsyncJobResponse(
+                client, cloudStackUrl, jobId, cloudStackUser);
+        while(response.getJobStatus() == CloudStackQueryJobResult.PROCESSING) {
+            if (countTries >= MAX_TRIES) {
+                throw new TimeoutCloudstackAsync(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
             }
-            throw new TimeoutCloudstackAsync(String.format(Messages.Exception.JOB_TIMEOUT, jobId));
-        } else {
-            throw new UnexpectedException();
+            sleepThread();
+            response = getAsyncJobResponse(client, cloudStackUrl, jobId, cloudStackUser);
```

Comment:
do we really need to wait the request to finish? Can't we be asynchronous either? If we could, we would answer OK to the user and he would notice if the request was successfully served on the dashboard... I'm saying this because I don't know how it used to work in the previous version.

Reply:
We may talk more about this topic because this affects many plugins. In this case, I've only refactored the old code. Someone made this decision architectural previously.

Reply:
ok, no need to change it now